op-conductor/health/monitor.go

```diff
@@ -117,7 +117,16 @@ func (hm *SequencerHealthMonitor) loop() {
 		case <-ticker.C:
 			err := hm.healthCheck()
 			hm.metrics.RecordHealthCheck(err == nil, err)
-			hm.healthUpdateCh <- err
+			// Ensure that we exit cleanly if told to shutdown while still waiting to publish the health update
+		loop:
+			for {
+				select {
+				case hm.healthUpdateCh <- err:
+					break loop
+				case <-hm.done:
+					break loop
+				}
+			}
```

Comment:
```suggestion
			select {
				case hm.healthUpdateCh <- err:
					continue
				case <-hm.done:
					return
			}
```

I don't think you need the loop here - you can just select. We can also return early here rather than handling the `hm.done` on the next outer loop iteration.

Reply:
I think without the loop we would drop the health update if the channel is already full which probably isn't what we want - we would update it again on the next tick but it would add latency.  I suspect the broader fix is to store the current health status in an `Atomic` of some kind and use a signal to notify subscribers when it changes rather than trying to write it to a channel. That way it wouldn't matter if there were no subscribers or many subscribers and they'd always get the latest status without blocking.

Reply:
Returning early definitely makes sense.

Reply:
Why would it drop the health update? Go will only make the send non-blocking if there's a `default` case in the switch, see https://gobyexample.com/non-blocking-channel-operations for the examples.

Reply:
Oh yeah good point.